xbmc/cores/AudioEngine/Engines/ActiveAE/ActiveAE.cpp

```diff
@@ -1949,7 +1949,7 @@ bool CActiveAE::RunStages()
             {
               // copy the samples into the viz input buffer
               CSampleBuffer *viz = m_vizBuffersInput->GetFreeBuffer();
-              int samples = std::min(512, out->pkt->nb_samples);
+              int samples = out->pkt->nb_samples;
```

Comment:
I've left the "automatic" FFT as the old size.
If you want  a longer FFT then you can handle that inside the viz.
If any vis wants <= 512 words of data, then only use the first 512 words. No need to force that on everyone.